packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/calibrate-utils.ts

```diff
@@ -29,3 +30,64 @@ export const mergeResults = (
 	}
 	return { result, resultSummary };
 };
+
+/**
+	* Get the mean absolute error from a provided source truth and a simulation run.
+	* Utilied in calibration for charts
+	* Assume that simulationData is in the form of pyciemss
+			states end with _State
+			The timestamp column is titled: timepoint_id
+	* Assume that the mapping is in the calibration form:
+			Ground truth will map to datasetVariable
+			Simulation data will map to modelVariable AND not include _State
+ * transform data, utilize mae, return mean aboslute error for charts.
+ * Note: This will only compare rows with the same timestep value.
+ */
+export async function getErrorData(groundTruth: DataArray, simulationData: DataArray, mapping: CalibrateMap[]) {
+	const errors: DataArray = [];
+	const pyciemssMap = await parsePyCiemssMap(simulationData[0]);
```

Comment:
Pretty sure that `parsePyCiemssMap` does not need to be async, because it is just doing string twiddlings, and thus `getErrorData` does not nee to be async, and so forth.